cmd/dockerd/hack/malformed_host_override_test.go

```diff
@@ -72,26 +57,28 @@ func BenchmarkWithHack(b *testing.B) {
 	read := make([]byte, 4096)
 	b.SetBytes(int64(len(req) * 30))
 
-	l := MalformedHostHeaderOverrideConn{client, true}
 	go func() {
 		for {
 			if _, err := srv.Write(req); err != nil {
 				srv.Close()
 				break
 			}
-			l.first = true // make sure each subsequent run uses the hack parsing
 		}
 		close(done)
 	}()
 
 	for i := 0; i < b.N; i++ {
 		for i := 0; i < 30; i++ {
-			if n, err := l.Read(read); err != nil && err != io.EOF {
+			n := -1
+			if n, err := client.Read(read); err != nil && err != io.EOF {
+				b.Fatalf("read: %d - %d, err: %v\n%s", n, len(req), err, string(read[:n]))
+			}
+			if n, err := stripInvalidHeader(n, read); err != nil && err != io.EOF {
```

Comment:
Not sure I'm following why this is added.

Reply:
Maybe time is too long to remember the reason.